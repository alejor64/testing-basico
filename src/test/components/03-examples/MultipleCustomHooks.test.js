import React from 'react'
import {shallow} from 'enzyme'
import '@testing-library/jest-dom'
import MultipleCustomHooks from '../../../components/03-examples/MultipleCustomHooks'
import { useFetch } from '../../../hooks/useFetch'
import { useCounter } from '../../../hooks/useCounter'

jest.mock('../../../hooks/useFetch')
jest.mock('../../../hooks/useCounter')

describe('Prueba en el componente <MultipleCustomHooks />', () => {

    useCounter.mockReturnValue({
        counter: 10,
        increment: () => {}
    })

    test('Debe mostrarse correctamente', () => {
        useFetch.mockReturnValue({
            data: null,
            loading: true,
            error: null
        })

        const wrapper = shallow(<MultipleCustomHooks />)

        expect(wrapper).toMatchSnapshot()
    })

    test('Debe de mostrar la informaciÃ³n', () => {
        useFetch.mockReturnValue({
            data: [{
                author: 'Alejandro',
                quote: 'Hola mundo'
            }],
            loading: false,
            error: null
        })

        const wrapper = shallow(<MultipleCustomHooks />)
        // console.log(wrapper.html())

        const alerte = wrapper.find('.alert').exists()
        const pText = wrapper.find('.mb-0').text().trim()
        const footerTag = wrapper.find('footer').text().trim()

        console.log(footerTag)

        expect(alerte).toBe(false)
        expect(pText).toBe('Hola mundo')
        expect(footerTag).toBe('Alejandro')
    })
    
    
})
