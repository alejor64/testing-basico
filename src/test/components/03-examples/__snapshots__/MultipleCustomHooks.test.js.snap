// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en el componente <MultipleCustomHooks /> Debe mostrarse correctamente 1`] = `
<div>
  <h1>
    Breaking Bad Quotes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Siguiente Frase
  </button>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Anterior Frase
  </button>
</div>
`;
